pxtcompiler/emitter/emitter.ts

```diff
@@ -3310,13 +3404,13 @@ ${lbl}: .short 0xffff
                 return r;
             let tp = typeOf(e)
 
-            if (target.floatingPoint && (tp.flags & (TypeFlags.NumberLike | TypeFlags.Boolean)))
+            if (target.floatingPoint && (tp.flags & (TypeFlags.NumberLike | TypeFlags.Boolean | TypeFlags.BooleanLiteral)))
```

Comment:
at this point, we should really clean out the floating point stuff

Reply:
Maybe it needs a general cleanup but I'm convinced we should leave support for integer compilation for tiny targets (AVR, maybe some tiny ARMS)